akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefSourceSpec.scala

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import scala.concurrent.duration._
+import akka.stream.ActorFlowMaterializer
+import akka.stream.OverflowStrategy
+import akka.stream.testkit.AkkaSpec
+import akka.stream.testkit.StreamTestKit
+
+class ActorRefSourceSpec extends AkkaSpec {
+  implicit val mat = ActorFlowMaterializer()
+
+  "A ActorRefSource" must {
+
+    "emit received messages to the stream" in {
+      val s = StreamTestKit.SubscriberProbe[Int]()
+      val ref = Source.actorRef.to(Sink(s)).run()
+      val sub = s.expectSubscription
+      sub.request(2)
+      ref ! 1
+      s.expectNext(1)
+      ref ! 2
+      s.expectNext(2)
+      ref ! 3
+      s.expectNoMsg(500.millis)
+    }
+
+    "be useful together with buffer" in {
+      val s = StreamTestKit.SubscriberProbe[Int]()
+      // FIXME isn't there a race when starting up, if I send to the ref before the buffer has requested?
+      //       Do we ignore that and fix that with fusing, or shall I add an internal buffer to the ActorRefSourceActor?
```

Comment:
I guess the MaterializedValuePublisher I implemented in the other PR can be taken as a template, augmented with a buffer and proper request tracking to have a simple Fire-and-Forget Publisher if we want to pursue that path.